src/H3Bites.js

```diff
@@ -19,49 +21,49 @@ const H3Bites = props => {
     return 0;
   }
 
-  return Object.entries(groupBy(h3SoundBites(), "speaker"))
-    .sort((a, b) => compareAttributes(a, b, 0))
-    .map(([groupName, bitesData]) => {
-      const heading = <h2 style={{width: 'fit-content'}}>{groupName}</h2>;
-      const image = <img src={bitesData[0].imgSrc} style={{width: '10vw', height: 'auto'}}/>;
+  function dataToSearch(data) {
+    return [data.speaker.toLowerCase()].concat(
+      data.name.split(" ").map(d => d.toLowerCase())
+    );
+  }
 
-      const header = (
-        <div
-          style={{
-            alignItems: "flex-start",
-            display: "flex",
-            flexDirection: "row-reverse",
-            justifyContent: "flex-end",
-          }}
-        >
-          {heading}
-          {image}
-        </div>
-      );
+  function speakerNameMatchesSearchTerms(name) {
+    let nameMatches = searchTerms.every(st => name.toLowerCase().includes(st));
+    if (nameMatches) return true;
 
-      const bites = bitesData.sort((a, b) => compareAttributes(a, b, "name")).map((bite, i) => {
-        return (
-          <Bite
-            key={i}
-            name={bite.name}
-            src={bite.src}
-            speaker={bite.speaker}
-          />
-        );
+    return false;
+  }
+
+  function foundMatch(matchData) {
+    if (speakerNameMatchesSearchTerms(matchData[0])) return true;
+    return biteNameMatchesSearchTerms(matchData);
+  };
+
+  function biteNameMatchesSearchTerms(matchData) {
+    var lastFoundIndex;
+
+    return searchTerms.every((term) => {
+      return matchData.some((segment) => {
+        if (segment.includes(term)) {
+          if (lastFoundIndex && matchData.slice(lastFoundIndex).indexOf(segment) === -1) return false;
```

Comment:
This line hurt my head a bit. 
It's checking to make sure that your current search segment exists, yeah?

Reply:
To start, a good summary of the nested `#some` inside `#every` is "all (`#every`) of the search terms should be included in at least one (`#some`) of the match segments."

45. loops through search terms
46. for each term, it loops through each word in the name of the sound bite, and calls each word `segment` (ex. ["these", "are", "all", "segments"])
47. If the current segment includes the search term,
48. Yeah, this one is a little bit of a doozy. I explain more in the PR with the "Something is about to bust" example, but `lastFoundIndex` is the index of the segment that matched the last search term. It's used to keep track of where in the string the terms are found. `matchData.slice(lastFoundIndex)` is saying to look at the last segment that matched a search term and all the strings after it (but not the ones before, even if they might match). So if the name was "hello world", then "hell wo" would match but "world hello" would not, because the function would match "hell" (term) to "hello" (segment) and then set `lastFoundIndex` to 1, and so it would only look for a match for "wo" at position 1 in the segments array or later.

Does that explanation make it more or less confusing?